packages/time/src/unix-nano.ts

```diff
@@ -100,7 +99,9 @@ export class UnixNanoTimeStamp {
   }
 
   public toTimeStamp(): TimeStamp {
-    return new TimeStamp(new Date(Number(this.input)));
+    const date = new Date(Number(this.input) / Number(ms));
```

Comment:
To fix the "Invalid Date" problems I have divided each timestamp by ms so the number is small enough to work with Javascript dates and moment.